draft-ietf-anima-brski-prm.md

```diff
@@ -1694,117 +1751,153 @@ This results in a signed CA certificate(s) object (JSON-in-JWS), the CA certific
 
 ~~~~ aasvg
 +--------+    +------------+    +-----------+    +--------+    +------+
-| Pledge |    | Registrar- |    |  Domain   |    | Domain |    | MASA |
-|        |    |   Agent    |    | Registrar |    |   CA   |    |      |
+| Pledge |    | Registrar- |    |  Domain   |    |  Key   |    | MASA |
+|        |    |   Agent    |    | Registrar |    | Infra. |    |      |
 +--------+    +------------+    +-----------+    +--------+    +------+
  |                  |                 |                 |   Internet |
  ~                  ~                 ~                 ~            ~
 (6) Supply Voucher to Pledge
  ~                  ~                 ~                 ~            ~
  |                  |                 |                 |            |
  |<----opt. TLS---->|                 |                 |            |
- |<-----Voucher-----|                 |                 |            |
+ |<-----Voucher''---|                 |                 |            |
  |------vStatus---->|                 |                 |            |
  |                  |                 |                 |            |
  ~                  ~                 ~                 ~            ~
 ~~~~
 {: #exchangesfig_uc2_6 title="Voucher exchange" artwork-align="center"}
 
-### Request Artifact: Voucher
+The Registrar-Agent SHALL supply the voucher to the pledge via HTTP(S) POST to the pledge endpoint at `/.well-known/brski/svr`.
+The request body MUST contain the Registrar-Countersigned Voucher (Voucher') artifact previously acquired from the domain registrar as defined in {{rcv-artifact}}.
+In the request header, the Content-Type field MUST be set to `application/voucher-jws+json` as defined in [I-D.ietf-anima-jws-voucher] and the Accept field SHOULD be set to `application/jose+json`.
+
+Upon receiving the voucher, the pledge SHALL perform the signature verification in the following order:
+
+1. Verify the MASA signature as described in {{Section 5.6.1 of !RFC8995}} against the pre-installed manufacturer trust anchor (e.g., IDevID).
+2. Provisionally install the trust anchor contained in the `pinned-domain-cert` field of the voucher.
+3. Validate the registrar LDevID certificate received in the `agent-provided-proximity-registrar-cert` field of the previously received tPVR artifact using the pinned domain certificate.
+4. Verify registrar signature of the Voucher' artifact similar as described in {{Section 5.6.1 of !RFC8995}}, but use the pinned domain certificate instead of the MASA certificate for the verification.
+
+BRSKI-PRM instroduces Steps 3 and 4 to verify the provisionally accepted registrar LDevID certificate from the tPVR parameter `agent-provided-proximity-registrar-cert` (see {{agt_prx}}). TODO(what about agent-signed data verification?)
+If all steps stated above have been performed successfully, the pledge SHALL terminate the "provisional state" for the domain trust anchor and the registrar LDevID certificate.
```

Comment:
It is unclear, which registrar cert should be installed as trust anchor:

* the one from agent-provided-proximity-registrar-cert
* the one from the Voucher' additional signature
* both

Might be connected to the confusion about verifying the CA certs.

Reply:
the one in the voucher as this is the one that was acknowledged by the MASA.

Reply:
The one in the voucher is only the pinned domain certificate, which may be an issuing or root certificate up the chain.

"Supply Voucher to Pledge" however says (Draft 12 version):

> If all steps stated above have been performed successfully, the pledge SHALL terminate the "PROVISIONAL accept" state for the domain trust anchor and the registrar LDevID certificate.

Because of the confusing Step 3 and 4 in "Supply CA Certificates to Pledge", I interpreted this as if the registrar LDevID cert is also installed as trust anchor.

But it appears, the registrar LDevID cert is NOT INSTALLED; it is just about retroactively verify the input for the PVR was correct. I think this should then be explained in two separate sentences to avoid this confusion.

Reply:
Yes. You can view it as stepwise approach. The MASA give the blessing to the domain registrar, which in turn can then provide the CA certificates as new trust anchors

Reply:
Clarified that only pinned domain cert is installed.
Clarified agent-provided-proximity-registrar-cert is used similar to the provisional TLS session in BRSKI.